common/lostfiles.in

```diff
@@ -39,127 +35,197 @@ while getopts 'sz' OPTION; do
 done
 shift $((OPTIND -1))
 
-filter() {
-  # just filter
-  sed -e 's|^\t||;' | sort
+# Sorts a list of file names by size
+sort_by_size() {
+  tr '\n' '\0' | xargs -0 -n1 du -s | sort -rn -k1
 }
 
-filter_sort() {
-  sed -e '1d' | tr '\n' '\0' | xargs -0 -n1 du -s | sort -rn -k1
+# Converts a list of glob-like strings to a single regular expression
+# Only the * wildcard is supported. The rest of the characters are compared literally.
+miniglob_list_to_regex() {
+  arr=("$@")
+  str=$(IFS=$'\n' ; echo "${arr[*]}")
+  echo -n "$str" | sed 's/[^^*]/[&]/g; s/\^/\\^/g; s/\*/.*/g' | tr '\n' '|'
 }
 
-strict() {
-  export LC_ALL=C
-  comm -13 \
-    <(pacman -Qlq | sed -e 's|/$||' | sort -u) \
-    <(find /boot /etc /opt /srv /usr /var | sort -u)
+# Generates a regex to implement the strict excluded paths list.
+# Strict mode will only exclude the basic UNIX special paths (e.g. tempfs)
+make_strict_exclude_list_regex() {
+  exclude_list=(
+    "/dev"
+    "/home"
+    "/lost+found"
+    "/media"
+    "/mnt"
+    "/proc"
+    "/root"
+    "/run"
+    "/scratch"
+    "/sys"
+    "/tmp"
+  )
+  
+  exclude_list_regex=$(miniglob_list_to_regex "${exclude_list[@]}")
+  echo -n "( -regex \"|$exclude_list_regex|\" -prune )"
```

Comment:
Please note that this is not how the strict mode works since #40. Using an exclude list is not a good idea for it, and I believe this can only be slower than the current way of doing things.